addons/mail/models/mail_link_preview.py

```diff
@@ -142,3 +147,16 @@ def _gc_mail_link_preview(self):
             ('message_id', '=', False),
             ('create_date', '<', fields.Datetime.now() - timedelta(days=lifetime)),
         ], order='create_date ASC', limit=1000).unlink()
+
+    @api.model
+    def _init_ignore_list(self):
+        try:
+            base_url = urlparse(self.get_base_url())
+            port = base_url.port or False
+        except ValueError:
+            return
+        base_ignore_pattern = f".*{base_url.hostname}"
+        if port:
+            base_ignore_pattern = f"{base_ignore_pattern}:{port}"
+        base_ignore_pattern = f"{base_ignore_pattern}/(web|odoo)"  # ignore backend url by default
```

Comment:
It would make a little bit more sense to read as a single f-string with `...before{full_host}...after` rather than individual lines.
In particular it would read better if port was appended to host immediately, rather than appending port to "ignore pattern" (according to var name).